src/Plugin/media/Source/GoogleForm.php

```diff
@@ -56,6 +86,41 @@ public function getSourceFieldConstraints() {
     ];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+    $height_options = $this->getHeightFieldOptions();
+    $form['height_field_name'] = [
+      '#type' => 'select',
+      '#title' => $this->t('iFrame Height'),
+      '#default_value' => $this->configuration['height_field_name'],
+      '#empty_option' => $this->t('- Choose -'),
+      '#options' => $height_options,
+      '#description' => $this->t('Select the field that will store the height of the iFrame.'),
+      '#weight' => -99,
+    ];
+    return $form;
+  }
+
+  /**
+   * Get the field options for the iframe height.
+   *
+   * @return string[]
+   *   A list of field options for the media type form.
+   */
+  protected function getHeightFieldOptions() {
+    $fields = $this->entityFieldManager->getFieldDefinitions('media', 'google_form');
```

Comment:
Can't always assume the entity type name is `google_form`

Reply:
Can you help me understand what you mean here?  This is in the actual definition of the `google_form` bundle.  In what circumstance would we be using the Google Form media bundle when it wasn't called `google_form`?

Reply:
If you create another media type that is not named `google_form` but uses the google form setup. The bundle of the entity should never be hard coded in these classes.

Reply:
This now gets the bundle from the $form_state dynamically.  Thanks for the tip!